airflow/utils/dag_parsing_context.py

```diff
@@ -0,0 +1,73 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+from contextlib import contextmanager
+from enum import Enum
+from typing import NamedTuple, Optional
+
+
+class AirflowParsingContextType(Enum):
+    """Type of context of parsing the DAG."""
+
+    DAG_PROCESSOR = "DAG_PROCESSOR"
+    TASK_EXECUTION = "TASK_EXECUTION"
```

Comment:
I'm somewhat wary of this -- why do we want a DAG to ever know what mode it is in? Shouldn't the mode be one of

a) Give me all the dags in the file (in which case `context.dag_id` is None); or
b) Give me a specific DAG only?

Reply:
Sure. Good idea. I was not sure how to name it, but this seems better "split" than the 'execution" context.

Reply:
Ah sorry, I wasn't clear.

My suggestion is to remove the type entirely, and have the decision in dag files be `if context.dag_id is not None:`

Reply:
The decision is already base on that indeed (look at the example). So this is just "extra" metadata on the context. I think there is no harm to keep it?

Reply:
But yeah. you are right. YAGNI.

Reply:
Right. Sklightly smaller with the Enum type.